src/ConnectedMode/Persistence/BindingDto.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Persistence;
+
+[JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
+internal class BindingDto
+{
+    public string ServerConnectionId { get; set; }
+    public Uri ServerUri { get; set; } // left here for compatibility reasons
```

Comment:
Maybe you can group the three compatibility properties at the bottom and add a better comment explaining what does "compatibility reason" means. Example of a better comment "The information for these properties are now stored in a separate file for connections, but they are needed here for backward compatibility with previous binding formats"

Reply:
I see. But then maybe just improve the comment?